app/javascript/src/components/Player.tsx

```diff
@@ -88,6 +88,7 @@ const DebuggerStatSubtractButton: FC<{ onClick: () => void }> = props => (
 class Player extends React.Component<PlayerProps, PlayerState> {
   engine: DiagramEngine
   model: DiagramModel
+  debug: boolean
```

Comment:
I'm not that familiar with this codebase but is there a reason `debug` is an instance variable and not a state field? I see later on where it's toggled we set `this.debug = !this.debug` and then set reset some state fields to cause a re-render. It might be simpler to just have `state.debug` and set that along with the other state.

Reply:
No, there is not a reason other than my lack of expertise in React, so I will make that change.